packages/react-native-editor/src/index.js

```diff
@@ -1,161 +1,66 @@
+import { applyFilters, doAction } from '@wordpress/hooks';
+import { Component, cloneElement } from '@wordpress/element';
 
 /**
- * WordPress dependencies
+ * Internal dependencies
  */
-import {
-	validateThemeColors,
-	validateThemeGradients,
-} from '@wordpress/block-editor';
-import { unregisterBlockType, getBlockType } from '@wordpress/blocks';
-
-const reactNativeSetup = () => {
-	// Disable warnings as they disrupt the user experience in dev mode
-	// eslint-disable-next-line no-console
-	console.disableYellowBox = true;
-
-	I18nManager.forceRTL( false ); // Change to `true` to debug RTL layout easily.
-};
-
-const gutenbergSetup = () => {
-	const wpData = require( '@wordpress/data' );
-
-	// wp-data
-	const userId = 1;
-	const storageKey = 'WP_DATA_USER_' + userId;
-	wpData.use( wpData.plugins.persistence, { storageKey } );
-
-	setupApiFetch();
-
-	const isHermes = () => global.HermesInternal !== null;
-	// eslint-disable-next-line no-console
-	console.log( 'Hermes is: ' + isHermes() );
-
-	setupInitHooks();
-
-	const initializeEditor = require( '@wordpress/edit-post' ).initializeEditor;
-	initializeEditor( 'gutenberg', 'post', 1 );
-};
-
-const setupInitHooks = () => {
-	const wpHooks = require( '@wordpress/hooks' );
+import './globals';
+import { omit } from 'lodash';
+import initialHtml from './initial-html';
+import setupLocale from './setup-locale';
+import { getTranslation } from '../i18n-cache';
 
-	wpHooks.addAction(
-		'native.pre-render',
-		'core/react-native-editor',
-		( props ) => {
-			setupLocale( props.locale, props.translations );
+const registerGutenberg = ( beforeInitCallback ) => {
+	class Gutenberg extends Component {
+		constructor() {
+			super( ...arguments );
 
-			const capabilities = props.capabilities ?? {};
-			if (
-				getBlockType( 'core/block' ) !== undefined &&
-				capabilities.reusableBlock !== true
-			) {
-				unregisterBlockType( 'core/block' );
-			}
-		}
-	);
+			const parentProps = omit( this.props || {}, [ 'rootTag' ] );
 
-	// Map native props to Editor props
-	// TODO: normalize props in the bridge (So we don't have to map initialData to initialHtml)
-	wpHooks.addFilter(
-		'native.block_editor_props',
-		'core/react-native-editor',
-		( props ) => {
-			const { capabilities = {} } = props;
-			let {
-				initialData,
-				initialTitle,
-				postType,
-				featuredImageId,
-				colors,
-				gradients,
-				rawStyles,
-				rawFeatures,
-				galleryWithImageBlocks,
-				locale,
-			} = props;
+			// Setup locale
+			setupLocale( {
+				locale: parentProps.locale,
+				extraTranslations: parentProps.translations,
+				getTranslation,
+			} );
 
-			if ( initialData === undefined && __DEV__ ) {
-				initialData = initialHtml;
-			}
-			if ( initialTitle === undefined ) {
-				initialTitle = 'Welcome to Gutenberg!';
-			}
-			if ( postType === undefined ) {
-				postType = 'post';
+			if ( beforeInitCallback ) {
+				beforeInitCallback( parentProps );
```

Comment:
We provide a callback for executing code before initialization, this will be used by the Gutenberg Mobile setup.

Reply:
```suggestion
			if ( beforeInitCallback ) {
				beforeInitCallback();
```

The `parentProps` argument does not appear to be utilized anywhere. Did I miss something or should we remove the argument?

Reply:
Right, `parentProps` is not being used but since this will be the entry point for extra initialization code (like GB-mobile case), it would be interesting to keep it.